### PR TITLE
gh-120862: Simplify chained comparison

### PR DESCRIPTION
**Pull request by Wulian233**
It's a simple change that doesn't really change anything. **So I skipped the news**

* Issue: gh-120862

### CODE REVIEW

File: Lib/typing.py [hunk 1]
```diff
@@ -1628,8 +1628,7 @@ def __getitem__(self, params):
         msg = "Parameters to generic types must be types."
         params = tuple(_type_check(p, msg) for p in params)
         if (self._defaults
-            and len(params) < self._nparams
-            and len(params) + len(self._defaults) >= self._nparams
+            and len(params) < self._nparams <= len(params) + len(self._defaults)
```
Comment (nineteendo) [inline Lib/typing.py]:
```suggestion
            and 0 < self._nparams - len(params) <= len(self._defaults)
```

**Comment by eendebakpt** [comment]
In general PRs with only cosmetic changes are not accepted to avoid churn in the codebase. See https://devguide.python.org/triage/triage-team/. 
In this case I agree the logic of the code is unchanged, but I am not convinced the changes make the code easier to read.

**Comment by Wulian233** [comment]
Thank you very much for reviewing my PR :) I understand the principle of avoiding unnecessary changes to the codebase. However, I believe this change can bring several benefits:
 Although both expressions equivalent, but I think  this are more intuitively convey the relationship between the number of parameters



2. Reduced Evaluation: Though it doesn't mean much, chained comparison operators in Python evaluate the middle variable only once
[Python Docs](https://docs.python.org/3/reference/expressions.html#comparisons)  `except that each expression is evaluated at most once.`

 Thank you for your time and understanding :)

**Comment by AlexWaygood** [comment]
Thanks for the PR. Overall I find most of these changes to be a small improvement in readability, but I find the one in `typing.py` to be much less readable, I'm afraid: it's checking two distinct things there, and combining the two checks into one expression makes it harder for me to parse. I think there are some serious readability issues with some parts of `typing.py`, and I'd be willing to accept some PRs if they significantly improved that situation (I have some patches lying around locally), but I don't think this PR really addresses any of those issues in a significant way.

Maybe we could consider this PR without the change to `typing.py`, but overall I'm afraid I don't think the churn (and risk -- there's always some small risk entailed in changing any code) is justified by the small improvements in readability here, which are anyway subjective. So in this instance, I'll defer to our general policy, which is to not accept cosmetic PRs like this.

> 2\. Reduced Evaluation: Though it doesn't mean much, chained comparison operators in Python evaluate the middle variable only once

FWIW, I doubt that this has any significant impact on any of the bits of code you're changing here :-) if you _did_ want to make an argument based on performance, you'd have to present some benchmark results for us to look at

**Comment by Wulian233** [comment]
Thank you for your long comment, I will pay attention to these questions in the future, I have learned a lot from my own experience, thank you very much anyway☺️

**Pull request closed without merging**